runtime/centrifuge/src/migrations.rs

```diff
@@ -10,4 +10,119 @@
 // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 // GNU General Public License for more details.
 
-pub type UpgradeCentrifuge1024 = ();
+pub type UpgradeCentrifuge1024 = burn_unburned::Migration<super::Runtime>;
+
+// Copyright 2021 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+mod burn_unburned {
+	use sp_std::vec::Vec;
+
+	const LOG_PREFIX: &'static str = "BurnUnburnedMigration: ";
+	const LP_ETH_USDC: CurrencyId = CurrencyId::ForeignAsset(100_001);
+	const ETH_DOMAIN: Domain = Domain::EVM(1);
+
+	use cfg_types::{domain_address::Domain, tokens::CurrencyId};
+	use frame_support::traits::{
+		fungibles::Mutate,
+		tokens::{Fortitude, Precision},
+		OnRuntimeUpgrade,
+	};
+	use pallet_order_book::weights::Weight;
+	use sp_runtime::{
+		traits::{Convert, Get, Zero},
+		TryRuntimeError,
+	};
+
+	pub struct Migration<T>
+	where
+		T: orml_tokens::Config + frame_system::Config,
+		<T as orml_tokens::Config>::CurrencyId: From<CurrencyId>,
```

Comment:
NIT: Could be the above be
```suggestion
		T: orml_tokens::Config<CurrencyId = CurrencyId> + frame_system::Config,
```

 to avoid `into()`s?